sqlmesh/core/test/definition.py

```diff
@@ -544,10 +545,14 @@ def generate_test(
                     cte_query = cte_query.with_(prev.alias, prev.this)
 
                 cte_output = t.cast(SqlModelTest, test)._execute(cte_query)
-                ctes[cte.alias] = pandas_timestamp_to_pydatetime(
-                    cte_output.apply(lambda col: col.map(_normalize_dataframe)),
-                    cte_query.named_selects,
-                ).to_dict(orient="records")
+                ctes[cte.alias] = (
+                    pandas_timestamp_to_pydatetime(
+                        cte_output.apply(lambda col: col.map(_normalize_df_value)),
+                        cte_query.named_selects,
+                    )
+                    .replace({np.nan: None})
```

Comment:
i think these replaces will trigger a warning, i've seen it in sqlglot

Reply:
I'm not sure if this is the direction that warns, maybe it's just `None -> np.nan`. Looking into it now.

Reply:
Brought back the `np.nan -> None` mapping, couldn't repro https://github.com/TobikoData/sqlmesh/issues/1799. May leave the PR open for a bit to evaluate this change and check if there are warnings.